packages/kit/types/ambient.d.ts

```diff
@@ -257,6 +257,7 @@ declare module '$app/navigation' {
 	 * Typically, you might call this to speed up subsequent navigation.
 	 *
 	 * You can specify routes by any matching pathname such as `/about` (to match `src/routes/about.svelte`) or `/blog/*` (to match `src/routes/blog/[slug].svelte`).
+	 * Calling `preloadCode('*')` will preload all routes.
```

Comment:
We need to document that this is a footgun:

```suggestion
	 * Calling `preloadCode('*')` will preload all routes. Use this extremely sparingly — it's much better to use service workers to download code in the background. `preloadCode` should only be used for routes that you're confident the user is about to visit.
```

Writing that reanimates my hesitation though — 'here's an API, never use it' is a confusing message.

Reply:
The second added sentence applies to the function in general. People can use it as a foot gun either way if they decide to put a lot of stuff into the array. I don't think we should make valid use cases harder to achieve because someone else could go "oh yeah let's just load the whole 500 pages of my public facing website at once"

Reply:
What are these valid use cases?

Reply:
Preloading the whole code of your internal business app without needing to resort to service worker stuff

Reply:
If that's the use case — always loading the entire app — it would arguably be much better solved by disabling code-splitting: https://twitter.com/Rich_Harris/status/1583521793448775680. That way you get more efficient transfer/better compression and we could probably generate a smaller bundle in the first place, plus you'd get better visibility into how big your bundle is.

`codeSplit: false` is very much an 'I know what I'm doing' feature, whereas `preloadCode('*')` _feels_ like you're being smart by getting stuff ready before the user needs it, but in all likelihood you're actually making stuff worse.

I think there's a separate issue here which is that service workers shouldn't feel like dark magic. If people don't feel confident writing service worker logic then we should tackle that head-on:

```js
// src/service-worker.js
import { cacheBuild } from '$service-worker/utils';

cacheBuild(); 
```